From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LanzoZ <bhbxgre@gmail.com>
Date: Thu, 5 Feb 2026 22:33:00 +0700
Subject: [PATCH] Change BrandName in MinecraftServer


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index c2227d57ea2e2da537a313d4bfd2f8c7f776be64..fb212530bfb8b9a9268e7e8bdf70baa946186fd5 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1957,7 +1957,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return io.papermc.paper.ServerBuildInfo.buildInfo().brandName(); // Paper
+        return "HGPaperForkV1.1beta"; // Paper
     }
 
     public SystemReport fillSystemReport(SystemReport systemReport) {
