From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: LanzoZ <bhbxgre@gmail.com>
Date: Fri, 6 Feb 2026 10:36:49 +0700
Subject: [PATCH] LeafDecay Remake Mechanics0-1-1


diff --git a/net/minecraft/world/level/block/LeavesBlock.java b/net/minecraft/world/level/block/LeavesBlock.java
index ba2a28934e4dbd46686e86d5393f8da361796b43..ae317d892315e14bcf09b43d4f0ec94efc4000ba 100644
--- a/net/minecraft/world/level/block/LeavesBlock.java
+++ b/net/minecraft/world/level/block/LeavesBlock.java
@@ -146,7 +146,7 @@ public abstract class LeavesBlock extends Block implements SimpleWaterloggedBloc
 
         int distance = getDistanceAt(neighborState) + 1;
         if (distance != 1 || state.getValue(DISTANCE) != distance) {
-            scheduledTickAccess.scheduleTick(pos, this, 1); // สั่งให้อัปเดตใน tick ถัดไป
+            scheduledTickAccess.scheduleTick(pos, this, 5); // สั่งให้อัปเดตใน tick ถัดไป
         }
         return state;
     }
